src/backend/aspen/api/schemas/phylo_runs.py

```diff
@@ -26,32 +27,109 @@ def tree_type_must_be_supported(cls, value):
         return uppercase_tree_type
 
 
-class GroupResponseSchema(BaseResponse):
+class GroupResponse(BaseResponse):
     class Config:
         orm_mode = True
 
     id: int
     name: StrictStr
+    location: StrictStr
 
 
-class UserResponseSchema(BaseResponse):
+class UserResponse(BaseResponse):
     id: int
     name: str
 
 
-class PhyloRunResponseSchema(BaseResponse):
+class TreeResponse(BaseResponse):
+    # A root validator gets us access to all fields in a model, so if
+    # we need to generate a field as a composite of other fields, this
+    # is how we do it.
+    @root_validator(pre=False)
+    def _set_fields(cls, values: dict) -> dict:
+        if values["name"]:
+            return values
+
+        # Generate a nice tree name if one doesn't exist
+        json_filename = values["s3_key"].split("/")[-1]
```

Comment:
Most of the name-generation code in this PR is transliterated from [here](https://github.com/chanzuckerberg/aspen/blob/trunk/src/backend/aspen/app/views/phylo_trees.py#L35-L56)